lib/tasks/push_aliquot_data_to_warehouse.rake

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+namespace :pool_and_library_aliquots do
+  #  A rake task to be run to push all pool and library aliquots data to the warehouse for volume tracking
+  desc 'Push all pool and library aliquots data to the warehouse for volume tracking'
+
+  task push_data_to_warehouse: :environment do
+    puts '-> Pushing all pool and library aliquots data to the warehouse for volume tracking'
+
+    begin
+      # Following are the contexts where aliquots are created, out of which only the aliquots from libraries and pools (1,2,3) are used for volume tracking
+      # 1. When a library or pool is created, primary aliquot is created
+      # 2. When a library is used in pool or run, derived aliquot is created
+      # 3. When a pool is used in run, derived aliquot is created
+      # 4. When a request is imported into the reception, a primary aliquot is created
+      # 5. When a request is used in a pool or library, derived aliquot is created
+
+      # Filter all aliquots that are not from a Pacbio::Request
+      aliquots = Aliquot.where.not(source_type: 'Pacbio::Request')
+      Emq::Publisher.publish(aliquots, Pipelines.pacbio, 'volume_tracking')
```

Comment:
Could this publish method fail or return error messages?

Reply:
Yes, it fails in inner methods and raises an exception under different circumstances, such as when an Avro validation fails or an EMQ connection error occurs. In those cases, the exception propagates up to the publish method. Now, a rescue is added in the call within the rake task to catch the exception. Hope that helps.
[fb721ee](https://github.com/sanger/traction-service/pull/1388/commits/fb712ee66c790e9f41592e67dfaed783d598dd23)
[383218a](https://github.com/sanger/traction-service/pull/1388/commits/383218a1b54cc34c14ff714ff163d6cb080f5f0c)

Reply:
I assume `Aliquot.where.not` call returns a list of aliquots. Have we tested publishing a list of aliquots to the EMQ? I can't remember whether `tol-lab-share` is able to decode lists of messages.

Reply:
It returns an array, but these are not the direct arguments sent to EMQ. This is a wrapper call which calls other objects that convert these aliquots into the required message format, which will then be published one by one. Hope that makes sense.

Reply:
Yeah that makes sense 😄 Thanks.

Reply:
This is actually a very good opportunity to test the robustness of our messaging platform and our overall volume tracking solution. I suppose we could use training data, and check the UAT aliquots table since the training instance of our message consumer is pointed to UAT (mlwh doesn't have a training instance).